File: Arrays/Three way partitioning/partition.cpp
void threeWayPartition(vector<int>& array,int a, int b)
    {
        int n = array.size();
        int start=0;
        int end=n-1;
        int i=0;
        
        while(i<=end)
        {
            if(array[i]<a)
            {
                swap(array[start],array[i]);
                start++;
                i++;
            }
            
            else if(array[i]>b)
            {
                swap(array[end],array[i]);
                end--;
            }
            else
                i++;
        }
        //return -1;
    }

File: Arrays/Three way partitioning/partition.py
def threeWayPartition(self, array, a, b):
	    n=len(array)
	    start=0
	    end=n-1
	    i=0
	    
	    #partition around a
	    while i<=end:
	        if(array[i]<a):
	            array[start],array[i] = array[i],array[start]
	            start+=1
	            i+=1
	            
	        elif array[i]>b:
	            array[end],array[i] = array[i],array[end]
	            end-=1
	        else:
	            i+=1
	            
	    return -1
